### PR TITLE
Update ClickHouse scripts and update results

### PR DESCRIPTION
**Pull request by qoega**
Initial ClickHouse support was added to h2o-benchmark long time ago when some features could be not present in ClickHouse. Is was less maintainable so complex scripts were rewritten to a simpler templated way.

To make comparison close to the other systems
- Logic to write results to disk is made as here:
https://github.com/duckdblabs/db-benchmark/blob/49016238fdd78dcbc4e5119bb0cd34487ca5419c/duckdb-latest/join-duckdb-latest.R#L50 
- Logic to reflect NA in data in the schema is added as here:
https://github.com/duckdblabs/db-benchmark/blob/49016238fdd78dcbc4e5119bb0cd34487ca5419c/duckdb-latest/join-duckdb-latest.R#L53
- Cases for sorted data and unsorted data are testing sorted and unsorted schema in ClickHouse. Before that ClickHouse was tested to use only unsorted tables.

Although we specifically do not perform optimisations that require tuning settings to get better results for specific workload as this one:

https://github.com/duckdblabs/db-benchmark/blob/49016238fdd78dcbc4e5119bb0cd34487ca5419c/duckdb-latest/join-duckdb-latest.R#L63-L66
Notes: 
- ClickHouse has safety mechanism that does not allow to delete 50GB+ tables without additional confirmation. To not require configuration changes scripts now always set `force_drop_table` although it is needed only after J1_1e9* benchmark runs.
- Previously script stopped if clickhouse-server was already running. Not it just stops it in the beginning of benchmark run. This makes it much more easy for development purposes to run benchmark again after failure in previous one.
- Locally I do not use `clickhouse-mount-config.xml` but I left it for your current setup. Instead I recommend just to mount `/var/lib/clickhouse` to one of  nvme partitions.

### CODE REVIEW

File: clickhouse/exec.sh [hunk 1]
```diff
@@ -25,36 +25,92 @@ ch_active || exit 1
...
+HAS_NULL=$(clickhouse-client --query "SELECT splitByChar('_','$SRC_DATANAME')[4]>0 FORMAT TSV")
+IS_SORTED=$(clickhouse-client --query "SELECT splitByChar('_','$SRC_DATANAME')[5]=1 FORMAT TSV")
+ON_DISK=0
+
 if [ $1 == 'groupby' ]; then
   clickhouse-client --query "DROP TABLE IF EXISTS $SRC_DATANAME"
-  clickhouse-client --query "CREATE TABLE $SRC_DATANAME (id1 Nullable(String), id2 Nullable(String), id3 Nullable(String), id4 Nullable(Int32), id5 Nullable(Int32), id6 Nullable(Int32), v1 Nullable(Int32), v2 Nullable(Int32), v3 Nullable(Float64)) ENGINE = MergeTree() ORDER BY tuple();"
+  if [ $HAS_NULL -eq 1 ]; then
+    if [ $IS_SORTED -eq 1 ]; then
+      clickhouse-client --query "CREATE TABLE $SRC_DATANAME (id1 LowCardinality(Nullable(String)), id2 LowCardinality(Nullable(String)), id3 Nullable(String), id4 Nullable(Int32), id5 Nullable(Int32), id6 Nullable(Int32), v1 Nullable(Int32), v2 Nullable(Int32), v3 Nullable(Float64)) ENGINE = MergeTree() ORDER BY (id1,id2,id3,id4,id5,id6,v1,v2,v3);"
```
Comment (jangorecki) [inline clickhouse/exec.sh:41]: sortedness is AFAIR only by ID cols, not v*

Comment (qoega) [reply clickhouse/exec.sh:41]: I can change if you think it makes  more sense. IMO it should not help at all as first columns already define order

Comment (jangorecki) [reply clickhouse/exec.sh:41]: yes, but it is somehow misleading now by treating all columns the same. ID columns are dimensions, V columns are measures.

**Comment by jangorecki** [comment]
> Although we specifically do not perform optimisations that require tuning settings to get better results for specific workload as this one:
>
> https://github.com/duckdblabs/db-benchmark/blob/master/duckdb-latest/join-duckdb-latest.R#L63-L66

IMO that should not be allowed. The tools should detect that manually and provide optimization itself, not needing a user to tune config for particular data.

**Comment by qoega** [comment]
@Tmonster Can you review this PR?

**Comment by qoega** [comment]
> clickhouse/exec.sh lines 138-140 attempt to drop table present in the join benchmark

I'll check this one. I think some copy-paste garbage left. Originally right join tables were not cleared at all and I added drop logic there.

>Also, it seems many of commands are piped to stderr, which muddies the .err file generated

I think it always was that way, so I did not add any other logic. I will check what can be done and why it is in cerr.

> Specifically a 10 second difference for basic group by and 20 seconds for advanced group by when looking at the summaries.

I can see images where 50GB basic summary result is 36s and advanced 96s. So 10-20 seconds looks significant. I will run again on a clean setup(as I had previous vm stopped).

Can you check that your storage setup is correct and ClickHouse runs with nvme drive for `/var/lib/clickhouse` (or the other data path if you changed it)?
Minor versions should not change performance. So storage configuration is the easiest thing to check.

**Comment by qoega** [comment]
I checked what was wrong with `.err` - I just forgot to remove `set -x` that I added to debug scripts.

I ran locally again on a clean system, this is how basic groupby 50GB looks 
image

**Comment by Tmonster** [comment]
Thanks! I'll take a look. Should I try to use v23.10.4.25 or do you think v23.10.5.20 will have the same result?

**Comment by qoega** [comment]
I should be the same. I just installed latest deb package and it was updated recently. (Changelog is small and about bug fixes [here](https://github.com/ClickHouse/ClickHouse/blob/e664e66a9a5532bee7ff0533348b041fc2be556f/docs/changelogs/v23.10.5.20-stable.md?plain=1#L8))

**Comment by Tmonster** [comment]
Looks good! Was able to verify the results finally 
image

Will merge when CI finishes

**Pull request merged**